src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1448,18 +1459,33 @@ def _name_handler(self, node, parent):
         :returns: PSyIR representation of node
         :rtype: :py:class:`psyclone.psyGen.Reference`
         '''
-        if hasattr(parent.root, 'symbol_table'):
-            symbol_table = parent.root.symbol_table
-            try:
-                symbol_table.lookup(node.string)
-            except KeyError:
-                raise SymbolError(
-                    "Undeclared reference '{0}' found when parsing fparser2 "
-                    "node '{1}' inside '{2}'."
-                    "".format(str(node.string), repr(node), parent.root.name))
-
+        self._check_declared(node.string, parent)
         return Reference(node.string, parent)
 
+    def _check_declared(self, name, node):
+        '''
+        :param str: name of the symbol.
+        :param node: parent node of the PSyIR node we are constructing.
+        :type node: :py:class:`psyclone.psyGen.Node`
+
+        raises SymbolError: if there is one or more ancestor symbol \
+        table(s) and the name is not found in one of them.
```

Comment:
I think this docstring does not match the implementation, it should be:" if there is no symbol table or if the name is not found in any of the symbol tables in scope"

Reply:
The logic is that an exception is not raised if there are no symbol tables found. This is to support the case of the NEMO api which currently does not have symbol tables. I've made a TODO to point this out and have referenced the relevant issue (#500). I've also updated the wording to try to make it clearer.